### ISSUE TITLE
Menu items get localized in wrong language

### ISSUE THREAD

**Issue by jaap-karssenberg** [koehlma/pygtkspellcheck#23]
Somehow the menu items get locallized wrongly. Even though my system locale is set to english and translation language is english, menu items show up as spanish (?). No idea why.

![screenshot from 2015-09-26 11 06 49](https://cloud.githubusercontent.com/assets/1248286/10116757/00f38614-643f-11e5-970a-66237fbf7004.png)

**Comment by koehlma** [comment]
Okay, strange. I have no clue why this could have happened. I am using gedits translation files if they are present and the translation is done by gettext. What does `echo $LANG` output?

**Comment by jaap-karssenberg** [comment]
jaap@Fantomes:~/code/zim-trunk$ echo $LANG
  en_US.UTF-8

Will try the snippet where you borrow from gedit in a separate instance.

Btw. in my application "_" is already mapped as a __builtin__ function and I'm distributing to platforms that do not have gedit. Would it be a suggestion to fallback to builtin if available ?

Will follow up, thanks,

Jaap

**Comment by jaap-karssenberg** [comment]
Weird, just running the snippet in python works fine. If I run the simple example, the Spanish translation shows up. Wondering if one of the imports messes with the localization settings.

    jaap@Fantomes:~/code/zim-trunk$ python
    Python 2.7.9 (default, Apr  2 2015, 15:33:21) 
    [GCC 4.9.2] on linux2
    Type "help", "copyright", "credits" or "license" for more information.
    >>> import gettext
    >>> gettext.find('gedit')
    >>> _ =     _gedit = gettext.translation('gedit', fallback=True).gettext
    >>> 
    >>> _('Languages')
    'Languages'
    >>> _('Suggestions')
    'Suggestions'


Probable should re-test all of my issues with latest version.

-- Jaap

**Comment by jaap-karssenberg** [comment]
OK, figured out what is the problem.

In my case gettext does not find the gedit translations. Gedit is installed - translations are not. Probably because I use vanilla English system settings.

The fallback to translation files specific for gtkspellcheck makes sense (ignore my previous comment on fall back). And they find the correct translation files.

However the ".mo"files in the package are screwed up - english is a copy of the spanish translation!

    jaap@Fantomes:~$ md5sum /usr/share/locale/es/LC_MESSAGES/pygtkspellcheck2.mo
    ef6a541d41195f2506c4e9efcd5b43bf  /usr/share/locale/es/LC_MESSAGES/pygtkspellcheck2.mo
    jaap@Fantomes:~$ md5sum /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck2.mo
    ef6a541d41195f2506c4e9efcd5b43bf  /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck2.mo

Issue is not in my system, the ".deb" files also contains the same files for both languages

    jaap@Fantomes:~/code$ dpkg-deb -c /var/cache/apt/archives/python-gtkspellcheck_3.0-1.1_all.deb  | grep ".mo$"
    -rw-r--r-- root/root      1627 2013-05-19 01:27 ./usr/share/locale/es/LC_MESSAGES/pygtkspellcheck2.mo
    -rw-r--r-- root/root      1627 2013-05-19 01:27 ./usr/share/locale/en/LC_MESSAGES/pygtkspellcheck2.mo
    -rw-r--r-- root/root      1701 2013-05-19 01:27 ./usr/share/locale/de/LC_MESSAGES/pygtkspellcheck2.mo


Not sure where the issue comes from - all I can say is that in the current git tree the ".mo" files are sane.

Long story short: the deb package needs an update. Will open issue in the debian bug tracker as well referring here.

Cheers!

Jaap

**Comment by koehlma** [comment]
Thanks for this great analysis of the problem. I don't know what went wrong with those packages either. Greetings, Maximilian.

**Comment by koehlma** [comment]
I am working together with @BenWiederhake to fix this problem: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=830509

**Comment by BenWiederhake** [comment]
As of yet, I simply requested action from the previous maintainers.  It'll be a while until I can get the MIA-process rolling, or even start packaging myself.

**Comment by koehlma** [comment]
Finally the newest version is available inside Debian testing. Since this is not an upstream bug please report it to the maintainers of the Debian package if it continues to exist.

**Comment by vossenjp** [comment]
This just happened to me on one Mint-18.3 workstation, but not on another.  The bad one had the bad file with the ES md5sum:
```
[jp@ringo:T3:L1:C1610:J0:2019-09-23_11:57:44_EDT]
/home/jp$ md5sum /usr/share/locale/e[ns]/LC_MESSAGES/pygtkspellcheck3.mo 
ef6a541d41195f2506c4e9efcd5b43bf  /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck3.mo
```

The good one had no file at all.  **So for me the fix was this, then restart Zim:**
```
[jp@ringo:T3:L1:C1615:J0:2019-09-23_12:04:40_EDT]
/home/jp$ sudo mv /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck3.mo /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck3.mo.BAD_Spanish
```

Oddly, that bad file is owned by `python3-gtkspellcheck` which is installed on both machines.  Bad:
```
[jp@ringo:T3:L1:C1612:J0:2019-09-23_12:01:18_EDT]
/home/jp$ dpkg-query --search /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck3.mo
python3-gtkspellcheck: /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck3.mo

[jp@ringo:T3:L1:C1613:J0:2019-09-23_12:03:13_EDT]
/home/jp$ dpkg-query --list | grep '^.i' | sort | egrep -i gtkspellcheck | awk '{OFS = "-"; print $2,$3}'
python3-gtkspellcheck-3.0-1.1
```

Yet, good:
```
[jp@weber6:T28:L1:C5001:J0:2019-09-23_12:15:11_EDT]
/home/jp$ dpkg-query --list | grep '^.i' | sort | egrep -i gtkspellcheck | awk '{OFS = "-"; print $2,$3}'
python-gtkspellcheck-4.0.5-1
python3-gtkspellcheck-3.0-1.1

[jp@weber6:T28:L1:C5002:J0:2019-09-23_12:15:20_EDT]
/home/jp$ dpkg -L python3-gtkspellcheck | /usr/bin/less --no-init
      1 /.
      2 /usr
      3 /usr/lib
      4 /usr/lib/python3
      5 /usr/lib/python3/dist-packages
      6 /usr/lib/python3/dist-packages/pylocales
      7 /usr/lib/python3/dist-packages/pylocales/__init__.py
      8 /usr/lib/python3/dist-packages/pylocales/locales.db
      9 /usr/lib/python3/dist-packages/pylocales/locales.py
     10 /usr/lib/python3/dist-packages/pygtkspellcheck-3.0.egg-info
     11 /usr/lib/python3/dist-packages/gtkspellcheck
     12 /usr/lib/python3/dist-packages/gtkspellcheck/oxt_import.py
     13 /usr/lib/python3/dist-packages/gtkspellcheck/__init__.py
     14 /usr/lib/python3/dist-packages/gtkspellcheck/spellcheck.py
     15 /usr/share
     16 /usr/share/doc
     17 /usr/share/doc/python3-gtkspellcheck
     18 /usr/share/doc/python3-gtkspellcheck/README.md
     19 /usr/share/doc/python3-gtkspellcheck/copyright
     20 /usr/share/doc/python3-gtkspellcheck/changelog.Debian.gz
     21 /usr/share/locale
     22 /usr/share/locale/es
     23 /usr/share/locale/es/LC_MESSAGES
     24 /usr/share/locale/es/LC_MESSAGES/pygtkspellcheck3.mo
     25 /usr/share/locale/en
     26 /usr/share/locale/en/LC_MESSAGES
     27 /usr/share/locale/en/LC_MESSAGES/pygtkspellcheck3.mo
     28 /usr/share/locale/de
     29 /usr/share/locale/de/LC_MESSAGES
     30 /usr/share/locale/de/LC_MESSAGES/pygtkspellcheck3.mo

[jp@weber6:T28:L1:C5004:J0:2019-09-23_12:18:20_EDT]
/home/jp$ md5sum /usr/share/locale/*/LC_MESSAGES/pygtkspellcheck3.mo
md5sum: '/usr/share/locale/*/LC_MESSAGES/pygtkspellcheck3.mo': No such file or directory
```
